devito/ir/iet/analysis.py

```diff
@@ -231,20 +231,24 @@ def mark_halospot_useless(analysis):
     Update the ``analysis`` detecting the USELESS HaloSpots within ``analysis.iet``.
     """
     properties = OrderedDict()
-    for i, scope in analysis.scopes.items():
-        for hs in FindNodes(HaloSpot).visit(i):
-            # A HaloSpot is USELESS if *all* reads along the HaloSpot's `loc_indices`
-            # pertain to an increment expression
-            test = False
-            for f, hse in hs.fmapper.items():
-                for d, v in hse.loc_indices.items():
-                    readat = v.origin if d.is_Stepping else v
-                    reads = [r for r in scope.reads[f] if r[d] == readat]
-                    if any(not r.is_increment for r in reads):
-                        test = True
-                        break
-            if not test:
-                properties[hs] = USELESS
+    for hs, iterations in MapNodes(HaloSpot, Iteration).visit(analysis.iet).items():
+        # `hs` is USELESS if ...
+
+        # * ANY of its Dimensions turn out to be SEQUENTIAL
+        if any(SEQUENTIAL in analysis.properties[i]
+               for i in iterations if i.dim.root in hs.dimensions):
+            properties[hs] = USELESS
+            continue
+
+        # * ALL reads pertain to an increment expression
+        test = False
```

Comment:
is this permanent?

Reply:
wdym ? 
this is like "start assuming it's false; if not, will find out and set it true"